### PR TITLE
Add verifier tests

### PR DESCRIPTION
**Pull request by aljones15**
- Adds `vc-generator` that generates test data
- Adds `20-verify.js` suite
- Refactors tests to reduce clutter and improve naming conventions.
- Tests for invalid `proofPurpose` and invalid `created` now have valid signatures in their proof.

20220930_10h37m57s_grim
20220930_10h38m05s_grim

### CODE REVIEW

File: tests/20-verify.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          await verificationFail({credential, verifier});
+        });
+        it('If the "type" field is missing or invalid, a MALFORMED error ' +
+          'MUST be returned.', async function() {
+          this.test.cell = {columnId, rowId: this.test.title};
+          const credential = credentials.clone('issuedVc');
+          delete credential.proof.type;
+          await verificationFail({credential, verifier});
+        });
+        it('If the "created" field is missing or invalid, a MALFORMED error ' +
```
Comment (gannan08) [inline tests/20-verify.js:47]: Are we certain that `created` fields are required on proofs?

Comment (gannan08) [reply tests/20-verify.js:47]: This test can't be testing if the field is missing OR invalid. It is testing that the field is missing, full stop. If not, it is testing that the field exists and is invalid, full stop. Please revise the test description to reflect what is being tested.

File: tests/20-verify.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        const [verifier] = endpoints;
+        it('If the "proof" field is missing or invalid, a MALFORMED error ' +
+          'MUST be returned.', async function() {
+          this.test.cell = {columnId, rowId: this.test.title};
+          const credential = credentials.clone('issuedVc');
+          delete credential.proof;
+          await verificationFail({credential, verifier});
+        });
+        it('If the "type" field is missing or invalid, a MALFORMED error ' +
+          'MUST be returned.', async function() {
```
Comment (gannan08) [inline tests/20-verify.js:41]: This test can't be testing if the field is missing OR invalid. It is testing that the field is missing, full stop. If not, it is testing that the field exists and is invalid, full stop. Please revise the test description to reflect what is being tested.

File: tests/20-verify.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          delete credential.proof.type;
+          await verificationFail({credential, verifier});
+        });
+        it('If the "created" field is missing or invalid, a MALFORMED error ' +
+          'MUST be returned.', async function() {
+          this.test.cell = {columnId, rowId: this.test.title};
+          const credential = credentials.clone('noCreated');
+          await verificationFail({credential, verifier});
+        });
+        it('If the "verificationMethod" field is missing or invalid, ' +
```
Comment (gannan08) [inline tests/20-verify.js:53]: Please see earlier comments of testing whether a field is missing or invalid.

File: tests/20-verify.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+          'a MALFORMED error MUST be returned.', async function() {
+          this.test.cell = {columnId, rowId: this.test.title};
+          const credential = credentials.clone('noVm');
+          await verificationFail({credential, verifier});
+        });
+        it('If the "proofPurpose" field is missing or invalid, ' +
```
Comment (gannan08) [inline tests/20-verify.js:59]: See above.

File: tests/20-verify.js (hunk 1 above)
Comment (aljones15) [reply tests/20-verify.js:47]: Current version of DI spec does require `created`:

> created
> Required. The string value of an [[ISO8601](https://w3c.github.io/vc-data-integrity/#bib-iso8601)] combined date and time string generated by the [Proof Algorithm](https://w3c.github.io/vc-data-integrity/#create-proof-algorithm).

https://w3c.github.io/vc-data-integrity/#proofs

File: tests/20-verify.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    this.report = true;
+    this.rowLabel = 'Test Name';
+    this.columnLabel = 'Verifier';
+    this.implemented = [...match.keys()];
+    this.notImplemented = [...nonMatch.keys()];
+    for(const [columnId, {endpoints}] of match) {
+      describe(columnId, function() {
+      // wrap the testApi config in an Implementation class
+        const [verifier] = endpoints;
+        it('If the "proof" field is missing or invalid, a MALFORMED error ' +
```
[the remark this answers is not in the document]
Comment (aljones15) [reply tests/20-verify.js:33]: addressed here: https://github.com/w3c-ccg/di-eddsa-2022-test-suite/pull/4/commits/87e557e2f04d36626b6aa7db82c5f8e967b16502

File: tests/20-verify.js (hunk 3 above)
Comment (aljones15) [reply tests/20-verify.js]: addressed here: https://github.com/w3c-ccg/di-eddsa-2022-test-suite/pull/4/commits/87e557e2f04d36626b6aa7db82c5f8e967b16502

File: tests/20-verify.js (hunk 4 above)
Comment (aljones15) [reply tests/20-verify.js]: addressed here: https://github.com/w3c-ccg/di-eddsa-2022-test-suite/pull/4/commits/87e557e2f04d36626b6aa7db82c5f8e967b16502

File: tests/20-verify.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+          const credential = credentials.clone('noVm');
+          await verificationFail({credential, verifier});
+        });
+        it('If the "proofPurpose" field is missing or invalid, ' +
+          'a MALFORMED error MUST be returned.', async function() {
+          this.test.cell = {columnId, rowId: this.test.title};
+          const credential = credentials.clone('noProofPurpose');
+          await verificationFail({credential, verifier});
+        });
+        it('If the "proofValue" field is missing or invalid, ' +
```
[the remark this answers is not in the document]
Comment (aljones15) [reply tests/20-verify.js]: addressed here: https://github.com/w3c-ccg/di-eddsa-2022-test-suite/pull/4/commits/87e557e2f04d36626b6aa7db82c5f8e967b16502

File: tests/20-verify.js (hunk 1 above)
Comment (aljones15) [reply tests/20-verify.js]: Addressed here: https://github.com/w3c-ccg/di-eddsa-2022-test-suite/pull/4/commits/87e557e2f04d36626b6aa7db82c5f8e967b16502

File: tests/20-verify.js (hunk 2 above)
Comment (aljones15) [reply tests/20-verify.js:49]: addressed here: https://github.com/w3c-ccg/di-eddsa-2022-test-suite/pull/4/commits/836e6dadc05bb5d5ebfc17562c6e6364e462b7a6

**Review by gannan08 — APPROVED** [review]
lgtm

**Pull request merged**